### PR TITLE
[W3W] Add pending proposals

### PR DESCRIPTION
**Pull request by alexander-lsvk**
# Description

Resolves # (issue)

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

**Review by llbartekll — COMMENTED** [review]
there are unresolved comments in your previous PR
https://github.com/WalletConnect/WalletConnectSwiftV2/pull/744
wouldn't it be easier to reopen the last one so we can continue discussion?

File: Sources/WalletConnectSign/Services/HistoryService.swift [hunk 1]
```diff
@@ -17,15 +27,33 @@ final class HistoryService {
...
+            .compactMap { record -> RequestSubscriptionPayload<SessionType.ProposeParams>? in
+                guard let proposalParams = mapProposeParams(record) else {
+                    return nil
+                }
+                return RequestSubscriptionPayload(id: record.id, topic: record.topic, request: proposalParams, decryptedPayload: Data(), publishedAt: Date(), derivedTopic: nil)
+            }
+        
+        requestSubscriptionPayloads.forEach {
+            let proposal = $0.request
+            proposalPayloadsStore.set($0, forKey: proposal.proposer.publicKey)
```
Comment (flypaper0) [inline Sources/WalletConnectSign/Services/HistoryService.swift:44]: where are we reading from that store? Couldn't find

File: Sources/Auth/AuthClientFactory.swift [hunk 2]
```diff
@@ -1,15 +1,15 @@
 import Foundation
 
-public struct AuthClientFactory {
+import WalletConnectVerify
```
Comment (flypaper0) [inline Sources/Auth/AuthClientFactory.swift:3]: It will brake cocoapods

File: Sources/Web3Wallet/Web3WalletClient.swift [hunk 3]
```diff
@@ -190,19 +190,25 @@ public class Web3WalletClient {
...
+    /// Query pending proposals
+    /// - Returns: Pending proposals received from peer with `wc_sessionPropose` protocol method
+    public func getPendingProposals(topic: String? = nil) -> [(proposal: Session.Proposal, context: VerifyContext?)] {
+        signClient.getPendingProposals(topic: topic)
+    }
+    
     /// - Parameter id: id of a wc_sessionRequest jsonrpc request
     /// - Returns: json rpc record object for given id or nil if record for give id does not exits
-    public func getSessionRequestRecord(id: RPCID) -> Request? {
+    public func getSessionRequestRecord(id: RPCID) -> (request: Request, context: VerifyContext?)? {
```
Comment (flypaper0) [inline Sources/Web3Wallet/Web3WalletClient.swift:205]: What do you think about separate `getVerifyContext(id: RPCID)`.
To not affect current Client interface

File: Sources/WalletConnectSign/Sign/SignClientFactory.swift [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import Foundation
 
+import WalletConnectVerify
```
Comment (flypaper0) [inline Sources/WalletConnectSign/Sign/SignClientFactory.swift:3]: Will brake cocoapods

**Review by flypaper0 — APPROVED** [review]

**Review by llbartekll — APPROVED** [review]

**Pull request merged**